applications/scripts/deep_micrograph_cleaner/deep_micrograph_cleaner.py

```diff
@@ -0,0 +1,165 @@
+        
+        ## examples
+        self.addExampleLine('xmipp_deep_micrograph_cleaner -c path/to/inputCoords/ -o path/to/outputCoords -b $BOX_SIXE  -i  /path/to/micrographs/')
+        
+    def run(self):
+
+        args={}
+        gpusToUse="0"
+        if self.checkParam('-g'):
+          gpusToUse= self.getParam('-g')
+          if "None" in gpusToUse or "-1" in gpusToUse:
+            gpusToUse=None
+        args["gpus"]=gpusToUse
+
+        updateEnviron(gpusToUse)
+        
+        if self.checkParam('-i'):
+          mdObj= md.MetaData(os.path.expanduser( self.getParam('-i')))
+          args["inputMicsPath"]= []
+          for objId in mdObj:
+            args["inputMicsPath"]+= [mdObj.getValue(md.MDL_IMAGE, objId)]
+        else:
+          raise Exception("Error, input micrographs fnames are requried as argument")
+
+        if self.checkParam('-b'):
+          args["boxSize"]= self.getIntParam('-b')
+        else:
+          raise Exception("Error, box size in pixels is required as argument")
+
+        args["inputCoordsDir"], args["outputCoordsDir"], args["predictedMaskDir"]= None, None, None
+        if self.checkParam('-c'):
+          args["inputCoordsDir"]= os.path.expanduser( self.getParam('-c'))
+        if self.checkParam('-o'):
+          args["outputCoordsDir"]= os.path.expanduser( self.getParam('-o'))
+        if self.checkParam('--predictedMaskDir'):
+          args["predictedMaskDir"]= os.path.expanduser( self.getParam('--predictedMaskDir'))
+
+
+        args["downFactor"]= self.getDoubleParamWithDefault('-s', defaultVal=1.0)
+        args["deepThr"]= self.getDoubleParamWithDefault('--deepThr', conditionFun= lambda x: x <= 0 or x >= 1,
+                                                        defaultVal=None)
+
+        args["sizeThr"]= self.getDoubleParamWithDefault('--sizeThr', conditionFun= lambda x: x <= 0 or x >= 1,
+                                                        defaultVal=0.8)
+          
+        if args["inputCoordsDir"] is None and args["predictedMaskDir"] is None:
+          raise Exception("Either inputCoordsDir or predictedMaskDir (or both) must be provided")
+
+        if args["inputCoordsDir"] is not None and args["outputCoordsDir"] is None:
+          raise Exception("Error, if inputCoordsDir provided, then outputCoordsDir must also be provided")
+
+        if args["outputCoordsDir"] is not None and args["inputCoordsDir"] is None:
+          raise Exception("Error, if outputCoordsDir provided, then inputCoordsDir must also be provided")
+    
+
+        if self.checkParam('-d'):
+          args["deepLearningModel"]= self.getParam('-d')
+        else:
+          args["deepLearningModel"]=Plugin.getModel('deepMicrographCleaner', 'defaultModel.keras')
+
+        try:
+          from xmippPyModules.micrograph_cleaner_em.cleanMics import main
+        except ImportError as e:
+          print(e)
+          raise ValueError(BAD_IMPORT_MSG)
+        main(** args)
+ 
+
+def updateEnviron(gpus=None):
```

Comment:
I think we discussed this before, but I don't remember the conclusion.
However, you should not be setting visible devices programmatically.

Reply:
In this case, just pass the gpus="all" argument and let slurm mask CUDA_VISIBLE_DEVICES